apps/src/code-studio/components/libraries/LibraryCreationDialog.jsx

```diff
@@ -103,113 +78,112 @@ class LibraryCreationDialog extends React.Component {
+    return (
+      <LibraryPublisher
+        onPublishSuccess={libraryName =>
+          this.setState({
+            dialogState: DialogState.PUBLISHED,
+            libraryName: libraryName
+          })
+        }
+        libraryDetails={libraryDetails}
+        libraryClientApi={libraryClientApi}
+      />
+    );
   };
 
-  publish = () => {
-    let {
-      librarySource,
-      libraryName,
-      libraryDescription,
-      selectedFunctions,
-      sourceFunctionList
-    } = this.state;
-    let functionsToPublish = sourceFunctionList.filter(sourceFunction => {
-      return selectedFunctions[sourceFunction.functionName];
-    });
-
-    if (!(libraryDescription && functionsToPublish.length > 0)) {
-      this.setState({publishState: PublishState.INVALID_INPUT});
-      return;
+  render() {
+    let bodyContent;
+    const {dialogState, libraryName} = this.state;
+    const {dialogIsOpen} = this.props;
+    switch (dialogState) {
+      case DialogState.LOADING:
+        bodyContent = <LoadingDisplay />;
+        break;
+      case DialogState.PUBLISHED:
+        bodyContent = <SuccessDisplay libraryName={libraryName} />;
+        break;
+      case DialogState.CODE_ERROR:
+        bodyContent = <ErrorDisplay message={i18n.libraryCodeError()} />;
+        break;
+      case DialogState.NO_FUNCTIONS:
+        bodyContent = <ErrorDisplay message={i18n.libraryNoFunctionsError()} />;
+        break;
+      default:
+        bodyContent = this.displayContent();
     }
-
-    let libraryJson = libraryParser.createLibraryJson(
-      librarySource,
-      functionsToPublish,
-      libraryName,
-      libraryDescription
+    return (
+      <Dialog
+        isOpen={dialogIsOpen}
+        handleClose={this.handleClose}
+        useUpdatedStyles
+      >
+        <Body>
+          <PadAndCenter>
+            <div style={styles.libraryBoundary}>
+              <Heading1>{i18n.libraryExportTitle()}</Heading1>
+              {bodyContent}
+            </div>
+          </PadAndCenter>
+        </Body>
+      </Dialog>
     );
+  }
+}
 
-    // TODO: Display final version of error and success messages to the user.
-    this.props.clientApi.publish(
-      libraryJson,
-      error => {
-        console.warn(`Error publishing library: ${error}`);
-        this.setState({publishState: PublishState.ERROR_PUBLISH});
-      },
-      () => {
-        this.setState({publishState: PublishState.PUBLISHED});
-      }
-    );
-    dashboard.project.setLibraryName(this.state.libraryName);
-    dashboard.project.setLibraryDescription(libraryDescription);
-  };
+export class ErrorDisplay extends React.Component {
+  static propTypes = {message: PropTypes.string.isRequired};
 
-  displayError = errorMessage => {
-    return <div>{errorMessage}</div>;
-  };
+  render() {
+    const {message} = this.props;
+    return <div>{message}</div>;
+  }
+}
 
-  displayLoadingState = () => {
+export class LoadingDisplay extends React.Component {
```

Comment:
Could some of these be pure functional components instead of classes? https://towardsdatascience.com/react-best-practices-804def6d5215